### PR TITLE
Fixed bug where list of users in the chatbox would scroll horizontally if a name was too long. Fixed UI of server selector to match the rest of the UI.

### PR DESCRIPTION
**Pull request by mac10688**
Use ellipses for the username if the username extends the length of the parent.

### CODE REVIEW

File: widgets/cu-xmpp-chat/lib/index.css [hunk 1]
```diff
@@ -343,13 +343,16 @@
...
       overflow-y: auto;
       flex-grow: 1;
       margin: 0.5em 0; }
       .cse-chat .chat-frame .chat-tab-content.chat-users .chat-info-user {
         color: white;
         padding: 0.5em;
-        cursor: pointer; }
+        cursor: pointer;
+        text-overflow: ellipsis;
+        overflow-x: hidden; }
```
Comment (sysrage) [inline widgets/cu-xmpp-chat/lib/index.css]: Move the brace to a new line and aligned with the opener (yes I know it was wrong before your checkin).

**Comment by sysrage** [comment]
Looks good except for the picky style change mentioned above.

**Comment by mac10688** [comment]
That's from the compiled css. It seems silly to modify that file. If anything I'm not sure we should be tracking the lib files.

**Comment by sysrage** [comment]
Good call. Sorry I didn't catch that it was the compiled CSS. The 'lib' dir is now in .gitignore so that won't happen again. 👍

File: patcher/src/ts/sidebar/ServerSelect.tsx [hunk 2]
```diff
@@ -202,29 +123,47 @@ class ServerSelect extends React.Component<ServerSelectProps, ServerSelectState>
...
     return dictToArray;
   }
 
+  getSelectedIndex = () : number => {
+    const {currentServer} = this.props.serversState;
+    const {selectedChannel} = this.props.channelsState;
+    if (!currentServer && !selectedChannel) return 0;
+
+    return this.listAsArray.indexOf(this.listAsArray.find((i: any) => {
+      if (i.serverInfo && currentServer) return i.serverInfo.channelID == currentServer.channelID
```
Comment (Mehuge) [inline patcher/src/ts/sidebar/ServerSelect.tsx]: missing ;

File: patcher/src/ts/sidebar/ServerSelect.tsx [hunk 3]
```diff
@@ -202,29 +123,47 @@ class ServerSelect extends React.Component<ServerSelectProps, ServerSelectState>
...
+      else if (i.channelInfo && selectedChannel) return i.channelInfo.channelID == selectedChannel.channelID
+      return false;
+    }));
+  }
+  
   render() {
+
```
Comment (Mehuge) [inline patcher/src/ts/sidebar/ServerSelect.tsx]: remove extra blank line (nit-pick)
https://github.com/airbnb/javascript#whitespace--padded-blocks

File: patcher/src/ts/sidebar/ServerSelect.tsx [hunk 4]
```diff
@@ -202,29 +123,47 @@ class ServerSelect extends React.Component<ServerSelectProps, ServerSelectState>
...
     const {servers} = this.props.serversState;
     const {channels} = this.props.channelsState;
 
     if (!servers && !channels) return null;
 
     this.listAsArray = this.mergeServerChannelLists(servers, channels);
 
+    var list: any = null;
```
Comment (Mehuge) [inline patcher/src/ts/sidebar/ServerSelect.tsx]: Use let not var
https://github.com/airbnb/javascript#references--prefer-const

File: patcher/src/ts/sidebar/ServerSelect.tsx [hunk 5]
```diff
@@ -127,66 +74,40 @@ class ServerSelect extends React.Component<ServerSelectProps, ServerSelectState>
...
-       return (
-        <div className='server-select quickselect-active'>
-        <h5 className='label'>SELECT SERVER</h5>
-          <div>
-            <div className='server-details'>
-              <h5 className='server'>{server.channelInfo.channelName}</h5>
-            </div>
+  renderItem(item: any) {
+    if (item.serverInfo) {
+
```
Comment (Mehuge) [inline patcher/src/ts/sidebar/ServerSelect.tsx]: remove extra blank line (nit-pick)
https://github.com/airbnb/javascript#whitespace--padded-blocks

**Comment by mac10688** [comment]
Done.

**Pull request merged**